### ISSUE TITLE
Webpack Compilation error

### ISSUE THREAD

**Issue by smile18** [evanw/node-source-map-support#173]
Getting below error while compiling ... 

`WARNING in ./~/source-map-support/source-map-support.js
Module not found: Error: Can't resolve 'module' in 'D:\WebStormWork\myapp\node_modules\source-map-support'
 @ ./~/source-map-support/source-map-support.js 474:15-32
 @ ./~/bunyan/lib/bunyan.js
 @ ./~/ldapjs/lib/index.js
 @ multi webpack-dev-server/client
webpack: Compiled with warnings.`

**Comment by ChrisProlls** [comment]
Same as me, any ideas ?

**Comment by ordinaryorange** [comment]
I had the same error, but was able to remove the package depending on source-map-support, which then got rid of it. @smile18  If you cant remove bunyan what happens if you add the source-map-support package ?

**Comment by bzalasky** [comment]
It appears to be caused by these lines (~470 in source-map-support.js)...

    if (options.hookRequire && !isInBrowser()) {
        var Module;
        try {
            Module = require('module');
        } catch (err) {
        // NOP: Loading in catch block to convert webpack error to warning.
        ...
    }

Perhaps webpack 2 is treating this differently than webpack 1 did?

**Comment by jscinoz** [comment]
Webpack 2/3 will attempt to parse any and all ES dynamic import() / CommonJS require() calls present in input files. When the Webpack build is has `target: web`, neither module nor fs will be available, hence the require calls for each of these will fail to resolve and Webpack will fail the build.

To work around this, you can set node.fs and node.module both to "empty" in your Webpack configuration, e.g:
```js
{
  /* Other configuration keys */
  node: {
    fs: "empty",
    module: "empty",
  }
}
```

**Comment by Ventzy** [comment]
Any idea how to do that in angular-cli (without ng eject)?

**Comment by RaedsLab** [comment]
up?

**Comment by ramiy** [comment]
Had the same issue:

```
warning  in ./node_modules/vue-monaco/node_modules/monaco-editor/esm/vs/editor/common/services/editorSimpleWorker.js

Critical dependency: the request of a dependency is an expression


warning  in ./node_modules/source-map-support/source-map-support.js

Module not found: Error: Can't resolve 'module' in '.\node_modules\source-map-support'
```

**Solved** with Webpack configure mentioned above:

```
node: {
  fs: 'empty',
  module: 'empty',
},
```

Thanks @jscinoz !!!

**Comment by jasonwr** [comment]
Yeah this is an issue for me as well. Please fix your software @evanw

**Comment by LinusU** [comment]
> Please fix your software

Would be happy to accept a pull request to fix this @jasonwr ☺️

**Comment by jasonwr** [comment]
@LinusU I'm strapped for time at the moment. Lots of coding these days. Also I'd have to ramp up on the internals of this package before being able to submit a defect free PR.
